docs/pages/how-it-works/deploy-machines.mdx

```diff
@@ -0,0 +1,25 @@
+---
+title: How it works using VMs
+sidebar_label: Deploying machines
+---
+
+Machines to DevPod are the infrastructure that ultimately run your devcontainer. Some providers such as gcloud, aws, digitalocean are "machine" providers since
+they first setup a VM to deploy you container to.
+
+When DevPod starts a workspace, such as `devpod up`, devpod selects a provider based on your context and starts your devcontainer. If using a machine provider, devpod will check if it should create a VM first.
```

Comment:
> devpod selects a provider based on your context
That kind of sounds passive but it's driven by the provider the user currently selected.
Don't think the context is super important here, just that people can set a provider